src/plugins/saved_objects_management/public/lib/filter_query.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+export function filterQuery(allowedVals: string[], requestedVals: string[]): string[] {
+  const filteredVals = allowedVals.filter((val) => !requestedVals || requestedVals.includes(val));
+  return filteredVals;
```

Comment:
In my earlier comments, what I was trying to say was that if `requestedVals` is falsy (aka `!requestedVals` is true), the `filter` function will return true for all `val` values. In other words, when `requestedVals` is falsy, we needn't waste CPU on looping through the elements on the array.

```suggestion
  return requestedVals
    ? allowedVals.filter((val) => requestedVals.includes(val))
    : allowedVals;
```